packages/core/test/SignUp.ts

```diff
@@ -80,5 +84,38 @@ describe('Signup', function () {
                 'ReachedMaximumNumberUserSignedUp'
             )
         })
+
+        it('setUsername should succeed', async () => {
+            const tx = await unirepSocialContract.setUsername(
+                config.NUM_EPOCH_KEY_NONCE_PER_EPOCH,
+                oldUsername,
+                newUsername,
+                { value: DEFAULT_ATTESTING_FEE }
+            )
+            const receipt = await tx.wait()
+
+            expect(receipt.status).equal(1)
+            const isClaimed = await unirepSocialContract.usernames(newUsername)
+
+            expect(isClaimed, 'This username has not been updated').to.be.true
+        })
+
+        it('should fail if a username is double registered', async () => {
+            const accounts = await ethers.getSigners()
+            await unirepSocialContract
+                .connect(accounts[0])
+                .setUsername(
+                    config.NUM_EPOCH_KEY_NONCE_PER_EPOCH,
+                    newUsername,
+                    newUsername2,
+                    { value: DEFAULT_ATTESTING_FEE }
+                )
+                .then((t) => t.wait())
+        })
+
+        it('should fail if the old username is not freed', async () => {
```

Comment:
This test can be removed. It shouldn't be possible to make this case fail. If you want to keep it in user a fresh username for testing, don't use one that is shared with other tests.

Generally we don't want tests depending on each other because a single failure causes a cascade, and it introduces hard to diagnose bugs.